### PR TITLE
Support installing packaged extensions

### PR DESCRIPTION
**Pull request by msullivan**
An packaged extension is a .zip file with a MANIFEST.toml file.
Installing an extension consists of:
 1. Dropping the .zip file into the extensions/ search path of edgedb
 2. If present, installing the postgres extension files into
    the packaged postgres directory.

The installation is currently done by an `install-ext.py` python
script, which should get turned into part of the CLI.

On server startup, the server checks the search path for any .zip
files, checks the MANIFEST for the names and versions, and if the
package does not exist, runs the script to install it.

Work on https://github.com/edgedb/edgedb/issues/7733.

The major rationale of having a separate MANIFEST with name/version is
so that we don't need to parse the whole extension package in order to
determine the name and version of extension, but maybe that isn't
necessary. We could just parse to do it, or we could try getting those
from the filename?

### CODE REVIEW

**Comment by msullivan** [comment]
Another question: postgres extensions put files in both `lib` and `share`. Would it be worth invoking `pg_config` to get the `LIBDIR` and `SHAREDIR` separately? Currently we assume they are siblings.

**Comment by elprans** [comment]
> Another question: postgres extensions put files in both `lib` and `share`. Would it be worth invoking `pg_config` to get the `LIBDIR` and `SHAREDIR` separately? Currently we assume they are siblings.

Invoking separately is probably best.

**Comment by msullivan** [comment]
Maybe we should fully drop the MANIFEST, get the name and version from the file name, have the postgres extension always live in `pg/{lib,share}`, and run the `*.edgeql` files in alphabetical order?

**Comment by msullivan** [comment]
On a suggestion from @aljazerzen that I really liked, I turned `install-ext.py` from an "external script that we will rewrite in rust" into an `edgedb-load-ext` command that the CLI will invoke. This is nice because it reduces the amount of logic overall and narrows the interface between CLI and edgedb.

File: edb/load_ext/main.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    'include': 'includedir',
+}
+
+
+def install_pg_extension(
+    pkg: pathlib.Path,
+    pg_config: dict[str, str],
+) -> None:
+    with zipfile.ZipFile(pkg) as z:
+        with z.open('MANIFEST.toml') as m:
```
Comment (1st1) [inline edb/load_ext/main.py:53]: I'd consider renaming to `GEL_EXTENSION.toml`. cc @elprans

File: edb/tools/config.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+from __future__ import annotations
+
+
+import click
+
+from edb import buildmeta
+from edb.tools.edb import edbcommands
+
+
+@edbcommands.command("config")
```
Comment (1st1) [inline edb/tools/config.py:29]: can you add a help line for this command describing what it does

**Review by 1st1 — COMMENTED** [review]
Nice work!

File: edb/load_ext/main.py (hunk 1 above)
Comment (elprans) [reply edb/load_ext/main.py:53]: I don't see a need to be this specific, it's a generated file in a tarball.

Comment (msullivan) [reply edb/load_ext/main.py:53]: Currently it's a hand-written file, though it could also be generated by the build system

**Review by elprans — APPROVED** [review]

**Pull request merged**